vllm/entrypoints/openai/protocol.py

```diff
@@ -148,8 +152,15 @@ class ChatCompletionRequest(OpenAIBaseModel):
     temperature: Optional[float] = 0.7
     top_p: Optional[float] = 1.0
     tools: Optional[List[ChatCompletionToolsParam]] = None
-    tool_choice: Optional[Union[Literal["none"],
-                                ChatCompletionNamedToolChoiceParam]] = "none"
+    tool_choice: Optional[
+        Union[
+            Union[
+                Literal["none"],
+                Literal["auto"]
+            ],
+            ChatCompletionNamedToolChoiceParam
+        ]
+    ] = "none"
```

Comment:
I think we should add here

```python
parallel_tool_calls: Optional[bool] = False
```

as it is part of the tool API (https://platform.openai.com/docs/api-reference/chat/create#chat-create-tool_choice)

I run into errors because some client forcing  `False` resulting in a "BadRequestError: Error code: 400" [..] "'msg': 'Extra inputs are not permitted'"

In my opinion it is easier to be defensive here (maybe throw a warning that the option will be ignored in the current PR), rather than patching all clients and have crashes meanwhile

Reply:
Nice catch @RomainBrault - I can add it as an option and add the warning as you suggest.

Reply:
Please see f0569272 :)

Reply:
LGTM :)